### DISCUSSION TITLE
How to get DomContentload, Onload metric using puppeeter&lighthouse?

### DISCUSSION THREAD

**Discussion by qkrdlsrnjs11** [GoogleChrome/lighthouse#13796] [Q&A] [answered]
Hello, 

I want to get a DomContentload, Onload data when I go into web page.

I used a example code(below).
but json file does not included with 'DomContentload', 'Onload' data.
How can I get this metric data?

```
  const json = reportGenerator.generateReport(lhr, 'json');

  const audits = JSON.parse(json).audits; // Lighthouse audits
  console.log(audits)
  const first_contentful_paint = audits['first-contentful-paint'].displayValue;
  const total_blocking_time = audits['total-blocking-time'].displayValue;
  const time_to_interactive = audits['interactive'].displayValue;
```

<img width="405" alt="image" src="https://user-images.githubusercontent.com/12168038/160842660-6a367a66-dd06-4a30-a152-34a1227a39ef.png">
example code :  https://github.com/GoogleChrome/lighthouse/blob/master/docs/puppeteer.md

**Comment by adamraine** [comment, marked as answer]
The values are not displayed in the report, however they are stored under the hidden `metrics` audit:

```js
const metrics = lhr.audits['metrics'].details.items[0];
const onload = metrics.observedLoad;
const dcl = metrics.observedDomContentLoaded;
```
It should be noted that these values are *not throttled* by default and will not line up with the other Lighthouse metrics. If you need to get the throttled DCL and onload events, turn on DevTools throttling via the config:

```js
const config = {
	extends: 'lighthouse:default',
	settings: {
		throttlingMethod: 'devtools',
	}
}
```

For more on throttling: https://github.com/GoogleChrome/lighthouse/blob/master/docs/throttling.md.

**Reply by qkrdlsrnjs11** [reply]
@adamraine Thanks you. It works well :)

**Reply by qkrdlsrnjs11** [reply]
@adamraine I have one more question about "throttled".

1) "These values are not throttled" means same with this status? (screen shot)

* I just want to set fast environment. ( not 3G, offline ..) in this case, do I have to adjust your code(const config ~ devtools) ? 

<img width="594" alt="image" src="https://user-images.githubusercontent.com/12168038/161215969-268d4cf5-9fe8-48d7-868d-24295aed622a.png">

**Reply by adamraine** [reply]
> "These values are not throttled" means same with this status? (screen shot)

Lighthouse throttling settings are independent of the throttling settings you set in the DevTools performance panel. Even if you run Lighthouse from DevTools it will use its own throttling settings.

> I just want to set fast environment. ( not 3G, offline ..) in this case, do I have to adjust your code(const config ~ devtools) ?

Yes, you would need to modify throttling settings in the config settings. You can reference the [default settings](https://github.com/GoogleChrome/lighthouse/blob/2f0730013b6e55521825a5eca06a656b7097127d/lighthouse-core/config/constants.js#L22-L29) for this.

**Reply by qkrdlsrnjs11** [reply]
@adamraine I see thanks a lot.
